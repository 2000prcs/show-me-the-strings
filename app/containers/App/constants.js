/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 */


export const GET_STRINGS = 'Strings-App/App/GET_STRINGS';
export const GET_STRINGS_SUCCESS = 'Strings-App/App/GET_STRINGS_SUCCESS';
export const GET_STRINGS_ERROR = 'Strings-App/App/GET_STRINGS_ERROR';

