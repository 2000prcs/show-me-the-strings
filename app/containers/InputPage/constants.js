
// InputPage Constants

export const CHANGE_STRING = 'app/InputPage/CHANGE_STRING';
export const INSERT_STRING = 'app/InputPage/INSERT_STRING';
export const INSERT_ERROR = 'app/InputPage/INSERT_ERROR';
