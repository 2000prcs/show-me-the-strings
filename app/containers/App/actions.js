/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import { GET_STRINGS, GET_STRINGS_SUCCESS, GET_STRINGS_ERROR } from './constants';

/**
 * Fetch the strings, this action starts the request saga
 *
 * @return {object} An action object with a type of GET_STRINGS
 */
export function getStrings() {
  return {
    type: GET_STRINGS,
  };
}

/**
 * Dispatched when the strings are loaded by the request saga
 *
 * @param  {array} strings The strings data
 *
 * @return {object}      An action object with a type of GET_STRINGS_SUCCESS passing the repos
 */
export function stringsFetched(strings) {
  return {
    type: GET_STRINGS_SUCCESS,
    strings,
  };
}

/**
 * Dispatched when fetching the strings fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of GET_STRINGS_ERROR passing the error
 */
export function stringsFetchingError(error) {
  return {
    type: GET_STRINGS_ERROR,
    error,
  };
}

