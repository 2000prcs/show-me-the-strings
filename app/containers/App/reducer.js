/*
 * AppReducer
 *
 * The reducer takes care of our data. Using actions, we can change our
 * application state.
 * To add a new action, add it to the switch statement in the reducer function
 *
 * Example:
 * case YOUR_ACTION_CONSTANT:
 *   return state.set('yourStateVariable', true);
 */

import { fromJS } from 'immutable';

import { GET_STRINGS, GET_STRINGS_SUCCESS, GET_STRINGS_ERROR } from './constants';

// The initial state of the App
const initialState = fromJS({
  loading: false,
  error: false,
  strings: [],
});

function appReducer(state = initialState, action) {
  switch (action.type) {
    case GET_STRINGS:
      return state
        .set('loading', true)
        .set('error', false)
        .set('strings', false);
    case GET_STRINGS_SUCCESS:
      return state
        .set('strings', action.strings)
        .set('loading', false)
    case GET_STRINGS_ERROR:
      return state.set('error', action.error).set('loading', false);
    default:
      return state;
  }
}

export default appReducer;