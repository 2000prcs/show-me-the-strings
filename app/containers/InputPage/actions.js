import { CHANGE_STRING, INSERT_STRING, INSERT_SUCCESS, INSERT_ERROR } from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} string The new text of the input field
 *
 * @return {object}    An action object with a type of CHANGE_STRING
 */
export function changeCurrentString(string) {
  return {
    type: CHANGE_STRING,
    string,
  };
}

/**
 * Insert a new string to DB
 * *
 * @return {object}      An action object with a type of INSERT_STRING
 */
export function insertString() {
  return {
    type: INSERT_STRING,
  };
}

/**
 * Error handling while insering a string to DB
 * *
 * @return {object}       An action object with a type of INSERT_SUCCESS passing the error
 */
export function insertingDBSuccess() {
  return {
    type: INSERT_SUCCESS,
  };
}


/**
 * Error handling while insering a string to DB
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of GET_STRINGS_ERROR passing the error
 */
export function insertingDBError(error) {
  return {
    type: INSERT_ERROR,
    error,
  };
}
