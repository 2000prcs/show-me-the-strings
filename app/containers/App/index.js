/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import HomePage from 'containers/HomePage/Loadable';
import InputPage from 'containers/InputPage/Loadable';
import messages from '../HomePage/messages';
import Title from '../../components/Title';
import Wrapper from '../../components/Title/Wrapper';
import Footer from '../../components/Footer';

export default function App() {
  return (
    <div>
      <Wrapper>
        <Title>
          <FormattedMessage {...messages.header} />
        </Title>
      </Wrapper>
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/input" component={InputPage} />
      </Switch>
      <Footer />
    </div>
  );
}
