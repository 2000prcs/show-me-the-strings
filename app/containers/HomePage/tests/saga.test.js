/**
 * Tests for HomePage sagas
 */

import { call, put, takeLatest } from 'redux-saga/effects';

import { GET_STRINGS } from 'containers/App/constants';
import { stringsFetched, stringsFetchingError } from 'containers/App/actions';
import request from 'utils/request';

import getData, { fetchStrings, requestURL } from '../saga';


describe('getData Saga', () => {
  const getDataGenerator = fetchStrings();

  it('Should request strings', () => {
    expect(getDataGenerator.next().value).toEqual(call(request, requestURL));    
  });

  it('Should dispatch the stringsFetched action if it requests the data successfully', () => {
    const response = [
      {
        id: 1,
        string: 'String 1',
        createdat: '2018-08-09',
      },
      {
        id: 2,
        string: 'String 2',
        createdat: '2018-08-09',
      },
    ];
    expect(getDataGenerator.next(response).value).toEqual(put(stringsFetched(response)));
  });
  
  it('Should be done after the action', () => {
    expect(getDataGenerator.next().done).toEqual(true);
  });
});


describe('getData Saga', () => {
  const getDataGenerator = fetchStrings();
  it('Should call the stringsFetchingError action if the response errors', () => {
    getDataGenerator.next();
    expect(getDataGenerator.throw('error').value).toEqual(put(stringsFetchingError('error')));
  });

  it('Should be done after the action', () => {
    expect(getDataGenerator.next().done).toEqual(true);
  });
});

describe('getData Saga', () => {
  const getDataSaga = getData();

  it('Should start task to watch for GET_STRINGS action', () => {
    const takeLatestDescriptor = getDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(GET_STRINGS, fetchStrings));
  });
});