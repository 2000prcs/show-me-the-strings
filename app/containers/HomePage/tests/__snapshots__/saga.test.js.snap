// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData Saga should call the stringsFetchingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "Strings-App/App/GET_STRINGS",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getData Saga should dispatch the stringsFetched action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "Strings-App/App/GET_STRINGS",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
